flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/decorators/FlinkConfMountDecorator.java

```diff
@@ -156,6 +157,8 @@ private Pod decoratePod(Pod pod) {
         // Remove some configuration options that should not be taken to cluster side.
         clusterSideConfig.removeConfig(KubernetesConfigOptions.KUBE_CONFIG_FILE);
         clusterSideConfig.removeConfig(DeploymentOptionsInternal.CONF_DIR);
+        clusterSideConfig.removeConfig(RestOptions.BIND_ADDRESS);
+        clusterSideConfig.removeConfig(RestOptions.ADDRESS);
```

Comment:
Actually, same comment as for TmOptions#ADDRESS. This should be purely a client-side thing.